controllers/external/types.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package external
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// ReconcileOutput is a return type of the external reconciliation
+// of referenced objects
+type ReconcileOutput struct {
+	Result *unstructured.Unstructured
+	Err    error
```

Comment:
```suggestion
	Error    error
```

Reply:
Add godoc?

Reply:
Can we leave Err/Error out of this struct? It makes the error handling look and feel very different than other places.

Reply:
SGTM

Reply:
ok, will take it out